### PR TITLE
DT-1542 Surface offender MAPPA details

### PR DESCRIPTION
**Pull request by mikehalmamoj**
I have just defined the API for now, implementation to follow.

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/RiskResource.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            @ApiResponse(code = 403, message = "Forbidden - requires ROLE_COMMUNITY", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Offender not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+    @GetMapping(value = "offenders/crn/{crn}/risk/mappa")
+    public MappaDetails getOffenderMappaDetailsByOffenderId(final @PathVariable("crn") String crn) {
+        return mappaDetailsFor(offenderService.offenderIdOfCrn(crn));
+    }
+
+    private MappaDetails mappaDetailsFor(Optional<Long> maybeOffenderId) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/RiskResource.java:78]: Meh, could be final.

**Review by JamieDraperUK — APPROVED** [review]

File: src/main/java/uk/gov/justice/digital/delius/data/api/MappaDetails.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private LocalDate startDate;
+    @ApiModelProperty(value = "Next review date", example = "2021-04-27")
+    private LocalDate reviewDate;
+    @ApiModelProperty(value = "Team code", example = "N07SP1")
+    private String teamCode;
+    @ApiModelProperty(value = "Officer code", example = "N07A011")
+    private String officerCode;
+    @ApiModelProperty(value = "Probation area code", example = "N07")
+    private String probationAreaCode;
+}
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/data/api/MappaDetails.java:32]: maybe add the notes field?, also wonder if team, officer, probationArea would be better off being Team, Human etc, or at least KeyValues pairs so we get the description, e.g probationArea { code = ABC, description = NPS North West}

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/RiskResource.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        value = "*** BETA - coming soon *** Return the MAPPA details for an offender using offenderId", notes = "requires ROLE_COMMUNITY")
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden - requires ROLE_COMMUNITY", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Offender not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+    @GetMapping(value = "offenders/offenderId/{offenderId}/risk/mappa")
+    public MappaDetails getOffenderMappaDetailsByOffenderId(final @PathVariable("offenderId") Long offenderId) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/RiskResource.java:45]: feel free to ditch the offenderId version, I think we have observed that no one tends to use the offenderId version of any of our APIs

**Review by andymarke — COMMENTED** [review]
LGTM

**Pull request merged**